README.md

```diff
@@ -14,59 +16,207 @@ pkg> add https://github.com/heltonmc/Bessels.jl.git
+
+Bessel functions of the first kind, denoted as $J_{\nu}(x)$, can be called with `besselj(nu, x)` where `nu` is the order of the Bessel function with argument `x`. Routines are also available for orders `0` and `1` which can be called with `besselj0(x)` and `besselj1(x)`.
+
+```julia
+julia> v, x = 1.4, 12.3
+
+# generic call for any order v
+julia> besselj(v, x)
+-0.22796228516266664
+
+# v = 0
+julia> besselj0(x)
+0.11079795030758544
+
+# v = 1
+julia> besselj1(x)
+-0.1942588480405914
+```
+
+### Bessel Functions of the second kind
+
+$$ Y_{\nu} = \frac{J_{\nu} \cos(\nu \pi) - J_{-\nu}}{\sin(\nu \pi)}    $$
+
+Bessel functions of the second kind, denoted as $Y_{\nu}(x)$, can be called with `bessely(nu, x)`. Routines are also available for orders `0` and `1` which can be called with `bessely0(x)` and `bessely1(x)`.
+
+```julia
+julia> v, x = 1.4, 12.3
+
+# generic call for any order v
+julia> bessely(v, x)
+0.00911009829832235
+
+# v = 0
+julia> bessely0(x)
+-0.19859309463502633
+
+# v = 1
+julia> bessely1(x)
+-0.11894840329926633
+```
+
+### Modified Bessel functions of the first kind
+
+$$ I_{\nu} = \sum_{m=0}^{\infty} \frac{1}{m!\Gamma(m+\nu+1)}(\frac{x}{2})^{2m+\nu}    $$
+
+Modified Bessel functions of the first kind, denoted as $I_{\nu}(x)$, can be called with `besseli(nu, x)` where `nu` is the order of the Bessel function with argument `x`. Routines are also available for orders `0` and `1` which can be called with `besseli0(x)` and `besseli1`. Exponentially scaled versions of these functions $I_{\nu}(x) * e^{-x}$ are also provided which can be called with `besseli0x(nu, x)`, `besseli1x(nu, x)`, and `besselix(nu, x)`.
+
+```julia
+julia> v, x = 1.4, 12.3
+
+# generic call for any order v
+julia> besseli(v, x)
+23781.28963619158
+
+# exponentially scaled version
+julia> besselix(v, x)
+0.10824635342651369
+
+# v = 0
+julia> besseli0(x)
+25257.48759692308
+julia> besseli0x(x)
+0.11496562932068803
+
+# v = 1
+julia> besseli1(x)
+24207.933018435186
+julia> besseli1x(x)
+0.11018832507935208
+```
+
+### Modified Bessel Functions of the second kind
+
+$$ K_{\nu} = \frac{\pi}{2} \frac{I_{-\nu} - I_{\nu}}{\sin(\nu \pi)}    $$
+
+Modified Bessel functions of the second kind, denoted as $K_{\nu}(x)$, can be called with `besselk(nu, x)`. Routines are available for orders `0` and `1` which can be called with `besselk0(x)` and `besselk1`. Exponentially scaled versions of these functions $K_{\nu}(x) * e^{x}$ are also provided which can be called with `besselk0x(nu, x)`, `besselk1x(nu, x)`, and `besselkx(nu, x)`.
+
+```julia
+julia> v, x = 1.4, 12.3
+
+julia> besselk(v, x)
+1.739055243080153e-6
+
+julia> besselk0(x)
+1.6107849768886856e-6
+
+julia> besselk1(x)
+1.6750295538365835e-6
+```
+
+### Support for negative arguments
+
+Support is also provided for negative arguments and orders. However, if you use negative arguments, **real arguments could produce complex outputs**. See https://github.com/heltonmc/Bessels.jl/issues/30 for more details.
```

Comment:
I thought we didn't want to do this because it causes type stability issues?

Reply:
Let me submit a PR and fix this to return error

Reply:
Fixed in https://github.com/heltonmc/Bessels.jl/pull/32

Reply:
And now updated with error message here.